src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -136,6 +137,9 @@ public static bool IsMsBuildBasedProject(string projectFullPath)
                 // Disallow the import of targets/props from packages
                 argumentBuilder.Append(" /p:ExcludeRestorePackageImports=true ");
 
+                // Add all depenencies as top level restore projects if recursive is set
+                argumentBuilder.Append($" /p:RestoreNoRecurse={(!recursive).ToString()} ");
```

Comment:
i personally feel toggle parameters should only be passed only if you are trying to achieve a non-default behavior, but your call to make the change or not.